### PR TITLE
Added attach file functionality 

### PR DESCRIPTION
**Pull request by mshahrukh2**
Added a new method in Driver.java
            attachFile()

### CODE REVIEW

File: src/test/resources/com/znsio/sample/e2e/features/theapp.feature [hunk 1]
```diff
@@ -53,4 +53,11 @@ Feature: Scenarios for "The App"
...
-    Then "you" login again with invalid credentials - "znsio4", "invalid password"
\ No newline at end of file
+    Then "you" login again with invalid credentials - "znsio4", "invalid password"
+
+  #CONFIG=./configs/theapp_local_config.properties TAG="fileupload and @theapp" PLATFORM=web ./gradlew run
+  @android @web @fileupload
+  Scenario: Verify file upload
+    Given I am on file upload page
+    When I upload the "image" file
+    Then File must be uploaded successfully
```
Comment (anandbagmar) [inline src/test/resources/com/znsio/sample/e2e/features/theapp.feature:63]: Then File is uploaded successfully

File: src/test/resources/com/znsio/sample/e2e/features/ajio.feature [hunk 2]
```diff
@@ -0,0 +1,9 @@
+@ajio @prod @guestUser
+Feature: Ajio tests
+
+#  CONFIG=./configs/ajio_local_config.properties TAG="@ajio and @photosearch" PLATFORM=android ./gradlew run
+  @android @single-app @imagesearch
```
Comment (anandbagmar) [inline src/test/resources/com/znsio/sample/e2e/features/ajio.feature:5]: is this test for android and web?

File: src/test/resources/com/znsio/sample/e2e/features/theapp.feature [hunk 3]
```diff
@@ -53,4 +53,11 @@ Feature: Scenarios for "The App"
     And "you" login with invalid credentials - "znsio2", "invalid password" on "android"
     And "someoneelse" login with invalid credentials - "znsio3", "invalid password" on "android"
     When "I" login again with invalid credentials - "znsio3", "invalid password"
-    Then "you" login again with invalid credentials - "znsio4", "invalid password"
\ No newline at end of file
+    Then "you" login again with invalid credentials - "znsio4", "invalid password"
+
+  #CONFIG=./configs/theapp_local_config.properties TAG="fileupload and @theapp" PLATFORM=web ./gradlew run
+  @android @web @fileupload
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/resources/com/znsio/sample/e2e/features/theapp.feature:59]: This test is for only @web, will remove @android tag.

File: src/test/resources/com/znsio/sample/e2e/features/ajio.feature (hunk 2 above)
Comment (mshahrukh2) [reply src/test/resources/com/znsio/sample/e2e/features/ajio.feature:5]: This test is for @android only.

File: src/test/java/com/znsio/sample/e2e/screen/android/ajio/HomeScreenAndroid.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    public HomeScreenAndroid(Driver driver, Visual visually) {
+        this.driver = driver;
+        this.visually = visually;
+    }
+
+    @Override
+    public SearchScreen searchByImage(Map imageData) {
+        String sourceFileLocation = System.getProperty("user.dir") + imageData.get("IMAGE_FILE_LOCATION");
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/java/com/znsio/sample/e2e/screen/android/ajio/HomeScreenAndroid.java:40]: Done.

File: src/test/java/com/znsio/sample/e2e/screen/android/ajio/HomeScreenAndroid.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    private static final By byStartSearchBoxId = By.id("com.ril.ajio:id/llpsTvSearch");
+    private static final By byUploadPhotoButtonId = By.id("com.ril.ajio:id/layout_select_photo");
+    private static final By byImageDirectoryXpath = By.xpath("//android.widget.TextView[contains(@text, 'Images')]");
+    private static final By byImageXpath = By.xpath("//android.view.ViewGroup[contains(@content-desc,'Photo taken')][1]");
+    private static final By byDismissButtonId = By.id("com.ril.ajio:id/footer_button_2");
+    private static final By bySystemPermissionMessageId = By.id("com.android.permissioncontroller:id/permission_message");
+    private static final By byAllowButtonId = By.id("com.android.permissioncontroller:id/permission_allow_button");
+
+
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/java/com/znsio/sample/e2e/screen/android/ajio/HomeScreenAndroid.java]: Done

File: src/test/java/com/znsio/sample/e2e/businessLayer/theapp/FileUploadBL.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+    public FileUploadBL navigationToUploadScreen(){
+        FileUploadScreen.get().navigateToFileUplaodPage();
+        return this;
+    }
+    public FileUploadBL uploadFile(Map file){
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/java/com/znsio/sample/e2e/businessLayer/theapp/FileUploadBL.java:44]: Updated

File: src/test/java/com/znsio/sample/e2e/screen/android/ajio/ProductScreenAndroid.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        driver.waitTillElementIsPresent(byAddToCartButtonId).click();
+
+        visually.checkWindow(SCREEN_NAME, "Add to cart");
+        driver.waitTillElementIsPresent(byViewBagButtonXpath).click();
+        return CartScreen.get();
+    }
+
+    @Override
+    public String getProductName() {
+        LOGGER.info("getProductName");
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/java/com/znsio/sample/e2e/screen/android/ajio/ProductScreenAndroid.java:38]: Done

File: src/test/java/com/znsio/sample/e2e/screen/web/theapp/FileUploadScreenWeb.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+public class FileUploadScreenWeb extends FileUploadScreen {
+    private static final Logger LOGGER = Logger.getLogger(FileUploadScreenWeb.class.getName());
+    private final Driver driver;
+    private final Visual visually;
+    private final String SCREEN_NAME = FileUploadScreenWeb.class.getSimpleName();
+    private final By byFileUploadXpath= By.xpath("//a[@href=\"/upload\"]");
+    private final By byChosseFileXpath= By.xpath("//input[@name=\"file\"]");
+    private final By byUplaodButtonId= By.id("file-submit");
+    private final By byFileUploadMessageXpath= By.xpath("//div[@class=\"example\"]/h3");
+
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/java/com/znsio/sample/e2e/screen/web/theapp/FileUploadScreenWeb.java]: Updated

File: src/test/java/com/znsio/sample/e2e/screen/web/theapp/FileUploadScreenWeb.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+public class FileUploadScreenWeb extends FileUploadScreen {
+    private static final Logger LOGGER = Logger.getLogger(FileUploadScreenWeb.class.getName());
+    private final Driver driver;
+    private final Visual visually;
+    private final String SCREEN_NAME = FileUploadScreenWeb.class.getSimpleName();
+    private final By byFileUploadXpath= By.xpath("//a[@href=\"/upload\"]");
+    private final By byChosseFileXpath= By.xpath("//input[@name=\"file\"]");
+    private final By byUplaodButtonId= By.id("file-submit");
+    private final By byFileUploadMessageXpath= By.xpath("//div[@class=\"example\"]/h3");
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/java/com/znsio/sample/e2e/screen/web/theapp/FileUploadScreenWeb.java]: Updated to //h3[contains(text(),'File Uploaded!')].

File: src/test/resources/com/znsio/sample/e2e/features/theapp.feature (hunk 1 above)
Comment (mshahrukh2) [reply src/test/resources/com/znsio/sample/e2e/features/theapp.feature]: Done

File: src/test/resources/com/znsio/sample/e2e/features/theapp.feature (hunk 3 above)
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/test/resources/com/znsio/sample/e2e/features/theapp.feature]: Done

File: src/main/java/com/znsio/e2e/tools/Driver.java [hunk 10]
```diff
@@ -465,4 +465,12 @@ public void switchToParentTab() {
             throw new NoSuchElementException("No previous tab found.", e);
         }
     }
+
+    public void uploadFileInBrowser(String filePath, By locator) {
```
[the remark this answers is not in the document]
Comment (mshahrukh2) [reply src/main/java/com/znsio/e2e/tools/Driver.java:469]: Done

**Pull request merged**